Jenkinsfile

```diff
@@ -124,30 +117,60 @@ def getTests(String baseUrl) {
 
 def getScenarios(String baseUrl, String testId){
     def scenarios = [[Id: "-1", Name: "Default"]]
-    def isScenarioChoosen = "${params['Run with scenarios']}".toBoolean()
+    def isScenarioChoosen = "${params['Run with datasets']}".toBoolean()
     if (isScenarioChoosen) {
-        def jsonSlurper = new JsonSlurper()
         URL url = new URL("${baseUrl}/projects/${params['Project']}/tests/${testId}/scenarios")
-        HttpURLConnection conn = url.openConnection()
-        conn.setRequestMethod("GET")
-        conn.setDoInput(true)
-        conn.setRequestProperty("Authorization", "APIKEY $useMangoApiKey")
-        conn.connect()
-        String content
-        def responseCode = conn.responseCode
-        if (responseCode == 200) {
-            InputStream inputStream = conn.getInputStream()
-            content = inputStream.getText()
-        } else {
-            throw new Exception("Scenarios get request failed with code: ${responseCode}")
-        }
-        def scenarioPage = jsonSlurper.parseText(content)
+        String content = getConnectionContent(url, "Scenarios")
+        def scenarioPage = new JsonSlurper().parseText(content)
         if (scenarioPage != null) {
             scenarioPage.each { scenario ->
-                echo "${scenario.Name}"        
+                echo "${scenario.Name}"
                 scenarios << [Id: scenario.Id, Name: scenario.Name]
             }
         }
     }
     return scenarios
+}
+
+def getEnvId(String baseUrl) {
+    String cursor = ""
+    echo "Retrieving environment based on EnvName"
+    while(true) {
+        URL url = buildURL("${baseUrl}/projects/${params['Project']}/environments", [
+                q: params['Environment'].toString(),
+                cursor: cursor
+        ])
+        String content = getConnectionContent(url, "Environment");
+        def environments = new JsonSlurper().parseText(content)
+        for (environment in environments["Items"]) {
+            if (environment["Name"].toString() == "${params['Environment']}".toString()) {
+                return environment["Id"]
+            }
+        }
+        if (!environments.Info.HasNext){
+            break
+        }
+        cursor = environments.Info.Next
+    }
+    throw new NoSuchElementException("The ${params['Environment']} environment does not exists")
+}
+
+def getConnectionContent(URL url, String requestedFor) {
+    HttpURLConnection conn = url.openConnection()
+    conn.setRequestMethod("GET")
+    conn.setDoInput(true)
+    conn.setRequestProperty("Authorization", "APIKEY $useMangoApiKey")
+    conn.connect()
+    if (conn.responseCode == 200) {
+        return conn.getInputStream().getText()
+    }
+    throw new Exception("${requestedFor} GET request failed with code: ${conn.responseCode}")
+}
+
+def buildURL(String path, Map<String, String> queryParams) {
```

Comment:
this method should be renamed as it is appending queryParams to URL.
there should be no control statement without braces. Even if the control statement has no code in it. (line number 171)